java/debugger.jpda/src/org/netbeans/modules/debugger/jpda/models/ShortenedStrings.java

```diff
@@ -204,6 +209,9 @@ static String getStringWithLengthControl(StringReference sr) throws
         }
         String string = null;
         boolean isShort = true;
+        Utf16State backingState = Utf16State.Chars;
+        //I think we are going with 'true' by default
+        boolean isLittleEndian;// = true;
```

Comment:
Please move declaration of `isLittleEndian` to initialisation site in 307. I would suggest to make it a `Boolean`, so that it is clear, that it might contain bogus data/is not valid. I thing asking the VM once for endianess should not be to big an issue (there is cache tied to the target machine), but ok.